### PR TITLE
Fix window paths on Node and add root relative pathing support

### PR DESCRIPTION
**Pull request by Zyie**
This PR fixes 2 issues:

- #8566
- #8564

For root-relative paths I added a new options to the adapter called `getRootPath` as these will be different for different adapters.
For windows paths I changed the checks being made for check if a path is absolute or not.

Previous Behaviour:

- Windows Node: 
  - `basePath` = `/`
  - `rootPath` = N/A
- Unix Node: 
  - `basePath` = `/`
  - `rootPath` = N/A
- Browser: 
  - `basePath` = `document.baseURI || window.location.href`
  - `rootPath` = N/A

New Behaviour:

- Windows Node: 
  - `basePath` = `process.cwd()` e.g. `C:\\Users\\foo\\bar`
  - `rootPath` = `C:\\`
- Unix Node: 
  - `basePath` = `process.cwd()` e.g. `/Users/foo/bar`
  - `rootPath` = `''`
- Browser: 
  - `basePath` = `document.baseURI || window.location.href`
  - `rootPath` = `window.location.origin`

This change to `basePath` in node would be a breaking change but makes more sense now that `rootPath` is being used.

I do think this is a band-aid fix as I think that all path code should be split out of assets but that is a bigger change that could be done in a separate PR?

### CODE REVIEW

File: bundles/pixi.js-node/src/adapter/adapter.ts [hunk 1]
```diff
@@ -19,45 +19,50 @@ export const NodeAdapter = {
...
+        return '';
+    },
     fetch: (url: RequestInfo, options?: RequestInit) =>
     {
         const request = new Request(url, options);
 
-        // if url is not absolute path then use fs to attempt to read the url
-        try
+        // check if urls starts with http(s) as only these are supported by node-fetch
+        if ((/^http(s)*:/).test(request.url))
```
Comment (bigtimebuddy) [inline bundles/pixi.js-node/src/adapter/adapter.ts:39]: RegExp could be a little cleaner.
```suggestion
        if ((/^https?:/).test(request.url))
```

File: packages/assets/src/utils/url/makeAbsoluteUrl.ts [hunk 2]
```diff
@@ -18,15 +20,25 @@ let baseUrl: string;
  * If it's not, we convert it
  * @param url - The URL to test
  * @param customBaseUrl - The base URL to use
+ * @param customRootUrl - The root URL to use
  */
-export function makeAbsoluteUrl(url: string, customBaseUrl?: string): string
+export function makeAbsoluteUrl(url: string, customBaseUrl?: string, customRootUrl?: string): string
```
Comment (bigtimebuddy) [inline packages/assets/src/utils/url/makeAbsoluteUrl.ts:25]: This definitely needs some unit-tests to handle the different permutations of platforms and requests.

File: packages/assets/src/utils/url/makeAbsoluteUrl.ts [hunk 3]
```diff
@@ -2,14 +2,16 @@ import { settings } from '@pixi/settings';
 import { isAbsoluteUrl } from './isAbsoluteUrl';
 import { urlJoin } from './urlJoin';
 
-export function getBaseUrl(url: string): string
+export function removeUrlParams(url: string): string
```
Comment (bigtimebuddy) [inline packages/assets/src/utils/url/makeAbsoluteUrl.ts:5]: Also, some unit-tests here would be helpful.

File: bundles/pixi.js-node/src/adapter/adapter.ts [hunk 4]
```diff
@@ -19,45 +19,50 @@ export const NodeAdapter = {
...
+        return '';
+    },
     fetch: (url: RequestInfo, options?: RequestInit) =>
     {
         const request = new Request(url, options);
 
-        // if url is not absolute path then use fs to attempt to read the url
-        try
+        // check if urls starts with http(s) as only these are supported by node-fetch
+        if ((/^https?:/).test(request.url))
```
Comment (bigtimebuddy) [inline bundles/pixi.js-node/src/adapter/adapter.ts:39]: What about data-URIs?

File: packages/assets/src/utils/url/isAbsoluteUrl.ts [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
-    // Don't match Windows paths `c:\`
-    if ((/^[a-zA-Z]:\\/).test(url))
+    // If running in node then an absolute URL will start will either a `/` on unix or disk drive on windows
+    if (settings.ADAPTER.getNavigator().userAgent === 'node')
     {
-        return false;
+        if ((/^[a-zA-Z][a-zA-Z\d+\-.]*:/).test(url)) return true;
+        if (process.platform === 'win32') return (/^[a-zA-Z]:(\\|\/)/).test(url);
+
+        return url.startsWith('/');
```
Comment (bigtimebuddy) [inline packages/assets/src/utils/url/isAbsoluteUrl.ts:20]: This seems like not the right place for this code. Part of the rationale of having an adapter is that is can externalize platform specifics (i.e., we should not have `process.platform` references in assets).

 An alternative approach:

```js
return settings.ADAPTER?.isAbsoluteUrl(url) ?? (/^[a-zA-Z][a-zA-Z\d+\-.]*:/).test(url);
```

**Comment by Zyie** [comment]
closing in favor of #8605

**Pull request closed without merging**